ARTstor.js

```diff
@@ -1,131 +1,899 @@
+    var serviceRoot = getServerUrl(url) + "/secure/";
+    return serviceRoot;
+
+}
+
+function getFileRoot(url) {
+    var fileRoot = getServerUrl(url).substring(0, getServerUrl(url).lastIndexOf('/'));
+    return fileRoot;
+}
+
+/**
+    decodeSearchData is an helper function to process the search result page.
+    It converts the search url parameter into arrary of parameter values.
+
+    It converts
+    type=6&kw=stimson hall|all#and,1925|all&geoIds=&clsIds=&collTypes=&id=all&bDate=-2019&eDate=2030&dExact=1&prGeoId=&origKW=stimson hall|all#and,1925|all
+    to  {
+             "type": 6
+             "kw": "stimson hall|all#and,1925|all"
+             "geoIds": ""
+             "clsIds": ""
+             "collTypes": ""
+             "id": "all"
+             "bDate": "-2019"
+             "eDate": "2030"
+             "dExact": "1"
+             "prGeoId": ""
+             "origKW": "stimson hall|all#and,1925|all"
+         }
+**/
+function decodeSearchData(str) {
+    var searchParam = str.replace(/(&gt;)/g, ":");
+    var param = searchParam.split('&');
+    var searchData = new Object();
+    var sparam;
+    var id;
+    var value;
+    for (var i = 0; i < param.length; i++) {
+        sparam = param[i].split('=');
+        id = sparam[0];
+        value = sparam[1];
+
+        if (id === 'type') {
+            searchData[id] = parseInt(value);
+        } else {
+            searchData[id] = value;
+        }
+    }
+    return searchData;
+}
+
+var decodeMap = {
+    '20': ' ',
+    '21': '!',
+    '22': '"',
+    '23': '#',
+    '24': '$',
+    '25': '%',
+    '26': '&',
+    '27': "'",
+    '28': '(',
+    '29': ')',
+    '2A': '*',
+    '2B': '+',
+    '2C': ',',
+    '2D': '-',
+    '2E': '.',
+    '2F': '/',
+    '30': '0',
+    '31': '1',
+    '32': '2',
+    '33': '3',
+    '34': '4',
+    '35': '5',
+    '36': '6',
+    '37': '7',
+    '38': '8',
+    '39': '9',
+    '3A': ':',
+    '3B': ';',
+    '3C': '<',
+    '3D': '=',
+    '3E': '>',
+    '3F': '?',
+    '5B': '[',
+    //              '5C' : '\\',
+    '5D': ']',
+    '5E': '^',
+    '5F': '_',
+    '60': '`',
+    '7B': '{',
+    '7C': '|',
+    '7D': '}',
+    '7E': '~'
+};
+
+encodeMap = {
+    ' ': '20',
+    '!': '21',
+    '"': '22',
+    '#': '23',
+    '$': '24',
+    '%': '25',
+    '&': '26',
+    "'": '27',
+    '(': '28',
+    ')': '29',
+    '*': '2A',
+    '+': '2B',
+    ',': '2C',
+    '-': '2D',
+    '.': '2E',
+    '/': '2F',
+    '0': '30',
+    '1': '31',
+    '2': '32',
+    '3': '33',
+    '4': '34',
+    '5': '35',
+    '6': '36',
+    '7': '37',
+    '8': '38',
+    '9': '39',
+    ':': '3A',
+    ';': '3B',
+    '<': '3C',
+    '=': '3D',
+    '>': '3E',
+    '?': '3F',
+    '[': '5B',
+    // '\\' : '5C',
+    ']': '5D',
+    '^': '5E',
+    '_': '5F',
+    '`': '60',
+    '{': '7B',
+    '|': '7C',
+    '}': '7D',
+    '~': '7E'
+};
+
+/**
+    Converts two character number character to special character
+**/
+function decrypt(s) {
```

Comment:
Can you explain to me what kind of data you're trying to "decrypt" here? Could you show a possible input and and output?